docs/source/compression/compression_config_list.rst

```diff
@@ -1,11 +1,190 @@
+  i.e., for a forward function ``def forward(self, x: Tensor, y: Tensor, z: Any): ...``, ``_input_0`` or ``_input_x`` can be used to specify ``x`` to be compressed,
+  note that ``self`` will be ignored when count the position index.
+  Similarly, ``_output_{position_index}`` can be used to specify the output target if the output is a ``list/tuple``,
+  ``_output_{dict_key}`` can be used to specify the output target if the output is a ``dict``.
+- Parameters/Buffers: Directly using the attribute name to specify the target, i.e., ``weight``, ``bias``.
+
+target_settings
+^^^^^^^^^^^^^^^
+
+A ``dict`` of target settings, the format is ``{target_name: setting}``. Target setting usually configure how to compress the target.
+
+All other keys(except these six common keys) in a config will seems as a shortcut of setting key, and will apply to all targets selected in this config.
+
+.. Note:: Each compression target can only be configure once, re-configuration will not take effect.
+
+Pruning Setting Keys
+--------------------
+
+sparse_ratio
+^^^^^^^^^^^^
+
+A float number between 0. ~ 1., the sparse ratio of the pruning target or the overall sparse ratio group of pruning tagrets.
+For example, if the sparse ratio is 0.8, and the pruning target is a Linear module weight, 80% weight value will be masked after pruning.
+
+max_sparse_ratio
+^^^^^^^^^^^^^^^^
+
+A float number between 0. ~ 1., for each pruning target, the sparse ratio after pruning will not be larger than this number,
+that means at most masked ``max_sparse_ratio`` pruning target value.
+
+min_sparse_ratio
+^^^^^^^^^^^^^^^^
+
+A float number between 0. ~ 1., for each pruning target, the sparse ratio after pruning will not be lower than this number,
+that means at least masked ``min_sparse_ratio`` pruning target value.
+
+sparse_threshold
+^^^^^^^^^^^^^^^^
+
+A float number, different from the ``sparse_ratio`` configure a specific sparsity, ``sparse_threshold`` usually used in some adaptive sparse cases.
+The value range is different for different pruning algorithms, please reference the pruner document to see how to configure it.
+In general, the higher the threshold, the higher the final sparsity.
+
+global_group_id
+^^^^^^^^^^^^^^^
+
+All pruning targets that have same ``global_group_id`` will be treat as a whole, and the ``sparse_ratio`` will be distributed across pruning targets.
+That means each pruning target might have different sparse ratio after pruning, but the group sparse ratio will be the configured ``sparse_ratio``.
+
+Note that the ``sparse_ratio`` in the same global group should be the same.
+
+dependency_group_id
+^^^^^^^^^^^^^^^^^^^
+
+All pruning targets that have same ``dependency_group_id`` will be treat as a whole, and the positions the targets' pruned will be the same.
+For example, layer A and layer B have same ``dependency_group_id``, and they want to be pruned output channels, then A and B will be pruned the same channel indexes.
```

Comment:
do you assume A and B have the same shape? if not, what constraints are put on A and B?

Reply:
the description could be clearer:)

Reply:
they should have same prunable positions shape.
tensor A weight shape (100, 100) with block granularity (10, 10), tensor B weight shape (50, 50) with block granularity (5, 5), after granularity reduction, they have same prunable positions shape (10, 10).

Reply:
better to add this example here. by the way, is it necessary for A and B to have the same number of dimensions?